### ISSUE TITLE
findExportByName works strange on android 8.1.0

### ISSUE THREAD

**Issue by alset0326** [frida/frida#856]
Hi, I recently upgrade to android 8.1.0.
When I use function `findExportByName` it would somehow reopen a new lib for me and then everything goes wrong.

For example, before I ran frida, the maps looks like this,
```sh
taimen:/ # cat /proc/2659/maps | grep libj.so
d9f6e000-d9f72000 r-xp 00000000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
d9f72000-d9f73000 r--p 00003000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
d9f73000-d9f74000 rw-p 00004000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
```
And then I just run
```javascript
'use strict';
Module.findExportByName('libj.so', 'Java_an_droid_j_MainActivity_j')
```
Then the maps become to this
```sh
taimen:/ # cat /proc/2659/maps | grep libj.so
d9894000-d9898000 r-xp 00000000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
d9898000-d9899000 r--p 00003000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
d9899000-d989a000 rw-p 00004000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
d9f6e000-d9f72000 r-xp 00000000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
d9f72000-d9f73000 r--p 00003000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
d9f73000-d9f74000 rw-p 00004000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
```
Look, there is a new `libj.so` appeared and the function `findExportByName` return the address of the new lib which would never be called.
But if I just use `findBaseAddress` and calculate the address by myself, everything works fun.
I have debug a lot to finally find it. Can you fix this? Thanks a lot!

**Comment by SheilaLundin** [comment]
For now, I use
```
var _symbols = {};
var _adjust = 0;
var _adjusted = false;
function adjustBase(lib)
{
    if (!_adjusted) {
        _adjusted = true;

        var sdk = 0;
        Java.perform(function () {
            var version = Java.use("android.os.Build$VERSION");
            sdk = version.SDK_INT.value;
        });
        if (sdk < 24) return;

        var rs = Process.enumerateRanges('r-x');
        var bases = [];
        for (var i = 0; i < rs.length; i++) {
            if (rs[i].file != undefined && rs[i].file.path.indexOf(lib) != -1) {
                bases.push(rs[i].base);
            }
        }
        bases.sort();
        _adjust = base(lib).sub(bases[0]);
    }
}
function findSymbol(lib, exportName)
{
    adjustBase(lib);

    if (_symbols[exportName] != undefined) {
        return _symbols[exportName];
    }

    var symbol = Module.getExportByName(lib, exportName);

    symbol = symbol.add(_adjust);
    _symbols[exportName] = symbol;
    return symbol;
}
```

**Comment by alset0326** [comment]
@SheilaLundin Sure this works, but the bug still there

**Comment by SheilaLundin** [comment]
@alset0326 what's the app? I'll take a try

**Comment by enovella** [comment]
> Hi, I recently upgrade to android 8.1.0.
> Then the maps become to this
> 
> ```shell
> taimen:/ # cat /proc/2659/maps | grep libj.so
> d9894000-d9898000 r-xp 00000000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
> d9898000-d9899000 r--p 00003000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
> d9899000-d989a000 rw-p 00004000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
> d9f6e000-d9f72000 r-xp 00000000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
> d9f72000-d9f73000 r--p 00003000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
> d9f73000-d9f74000 rw-p 00004000 08:0d 2744326                            /data/app/an.droid.j-5QA1Lki96_U7H2Ya9-b0jw==/lib/arm/libj.so
> ```

I have been experimenting stuff like that. Probably it needs attention to find the issue.

**Comment by oleavr** [comment]
Fixed in frida/frida-gum@3d6cb1d623c8eba2982d24a01463621156b94d23. New release should be out soon – just needs some more testing first. (Help appreciated!)
